Add support for clang>=16

MariusDoe opened the pull request:
Fixes #693.

MariusDoe committed:
Fix parameter types for SoundPlugin

To enable compilation with clang>=16

Co-authored-by: Christoph Thiede <christoph.thiede@student.hpi.uni-potsdam.de>

MariusDoe committed:
Explicitly define implicitly declared function for unix DropPlugin

To enable compilation with clang>=16

Fixes #693.

Co-authored-by: Christoph Thiede <christoph.thiede@student.hpi.uni-potsdam.de>

MariusDoe committed:
Add extern function declarations for implicitly declared functions in XDisplayControlPlugin

To enable compilation with clang>=16

Co-authored-by: Christoph Thiede <christoph.thiede@student.hpi.uni-potsdam.de>

eliotmiranda reviewed (changes requested):
Hi, 
    in fixing things please move from int to sqInt, not from sqInt to int. The reasons are that
a) sqInt is guaranteed to be equal to the word size whereas typically int is only 32 bits and so too small on 64-bits to capture the full range of possible values
b) because sqInt is word size typically better/more natural code is generated on 64-bit platforms than for int

So please redo the changes for the SoundPlugin so that sqInt is used for all parameters, not int. This affects
platforms/Cross/plugins/SoundPlugin/SoundPlugin.h - does not need to change, *should not change*
platforms/unix/vm-sound-sndio/sqUnixSndioSound.c
platforms//unix/vm/SqSound.h - does need to change, and files including it will need to also (but I don't see any)

If any files affected need to get the typedef for sqInt they should include sqMemoryAccess.h which defines only basic types and memory access and so does not pose a hazard in including too much.
Thank you.

MariusDoe committed:
Avoid editing generated files

This accompanies VMConstruction-Plugins-XDisplayControlPlugin.oscog-mad.20, which will include the new header in the generated file.

MariusDoe commented:
Superseded by #719.
